src/Api/Vault/Controllers/SyncController.cs

```diff
@@ -90,9 +95,23 @@ await _providerUserRepository.GetManyOrganizationDetailsByUserAsync(user.Id,
 
         var userTwoFactorEnabled = await _userService.TwoFactorIsEnabledAsync(user);
         var userHasPremiumFromOrganization = await _userService.HasPremiumFromOrganization(user);
-        var response = new SyncResponseModel(_globalSettings, user, userTwoFactorEnabled, userHasPremiumFromOrganization, organizationUserDetails,
-            providerUserDetails, providerUserOrganizationDetails, folders, collections, ciphers,
-            collectionCiphersGroupDict, excludeDomains, policies, sends);
+        var managedByOrganizationId = await GetManagedByOrganizationIdAsync(user, organizationUserDetails);
+
+        var response = new SyncResponseModel(_globalSettings, user, userTwoFactorEnabled, userHasPremiumFromOrganization,
+            managedByOrganizationId, organizationUserDetails, providerUserDetails, providerUserOrganizationDetails,
+            folders, collections, ciphers, collectionCiphersGroupDict, excludeDomains, policies, sends);
         return response;
     }
+
+    private async Task<Guid?> GetManagedByOrganizationIdAsync(User user, IEnumerable<OrganizationUserOrganizationDetails> organizationUserDetails)
+    {
+        if (!_featureService.IsEnabled(FeatureFlagKeys.AccountDeprovisioning) ||
```

Comment:
You can use [`RequireFeatureAttribute`](https://github.com/bitwarden/server/blob/9dedaa5acf3c8b087c61d529017629e982b6262f/src/Core/Utilities/RequireFeatureAttribute.cs#L12) here instead of injecting the feature service and checking it in the endpoint's function body.

Thus will change the behavior of the endpoint - it'll throw an exception if the feature is off instead of returning null. If that's not okay here let me know!

Reply:
The [RequireFeatureAttribute](https://github.com/bitwarden/server/blob/9dedaa5acf3c8b087c61d529017629e982b6262f/src/Core/Utilities/RequireFeatureAttribute.cs#L12) blocks the entire endpoint when the feature flag is disabled, but that's not the behavior we need. Since we're just adding a value to an existing endpoint, we only want to query that value if the feature flag is enabled.

Reply:
Duh! I just woke up 🥱

Thanks Rui!